merchant-sdk/example-app/src/main/java/net/gini/android/merchant/sdk/exampleapp/invoices/ui/InvoicesViewModel.kt

```diff
@@ -56,47 +52,24 @@ class InvoicesViewModel(
 
     fun loadPaymentProviderApps() {
         viewModelScope.launch {
-            paymentComponent.loadPaymentProviderApps()
+            giniMerchant.loadPaymentProviderApps()
         }
     }
 
-    fun getPaymentReviewFragment(documentId: String) {
-        viewModelScope.launch {
-            LOG.debug("Getting payment review fragment for id: {}", documentId)
-
-            _paymentReviewFragmentFlow.value = PaymentReviewFragmentState.Loading
+    fun setSelectedInvoiceItem(invoiceItem: InvoiceItem) = viewModelScope.launch {
+        _selectedInvoiceItem.emit(invoiceItem)
+    }
 
-            val documentWithExtractions =
-                invoicesRepository.invoicesFlow.value.find { it.documentId == documentId }
+    fun startIntegratedPaymentFlow(documentId: String) {
+        _startIntegratedPaymentFlow.tryEmit(giniMerchant.getContainerFragment(documentId, paymentFlowConfiguration))
```

Comment:
[Suggestion] Since `getContainerFragment()` is synchronous we could avoid using the `startIntegratedPaymentFlow` and just return the fragment directly here.

Reply:
`InvoicesViewModel` is currently shared between `InvoicesActivity` and `InvoiceDetailsFragment`. The reason why I opted for a flow here is because the payment flow is started by tapping on the `Pay Now` button in `InvoiceDetailsFragment`, but `InvoicesActivity` is the one that handles adding and removing fragments from its container view. 
So `InvoiceDetailsFragment` -> asks for the fragment -> sends it over to `InvoicesActivity` -> shows it in its hierarchy.
Some sort of flow is needed in my opinion to signal to the activity that is needs to do its job, but it can be a simple void flow...and let the activity call `giniMerchant.getContainerFragment`. Would this be a simpler / more straightforward approach in your opinion?

Reply:
Thank you for the clarification. In that case it can remain as it is. 

Sidenote: in EC-15 we will have to [replace Invoices List and Invoice Details](https://ginis.slack.com/archives/C0736AKHH0T/p1719237916687839?thread_ts=1718790245.972319&cid=C0736AKHH0T) with an Orders Screen and an Order Details Screen. We could then have an OrdersActivity which uses Jetpack Navigation to manage the navigation between Orders Screen <-> Order Details Screen <-> PayFlowFragment. But we don't have to decide that now.